### PR TITLE
Add support for Apple file list types for validation

### PR DESCRIPTION
**Pull request by sandyydk**
Changes in this PR

- Add validator for Plist file types
- Test cases for Plist file type validation

Closes https://github.com/Boeing/config-file-validator/issues/66

### CODE REVIEW

**Comment by sandyydk** [comment]
@kehoecj Tried to check plist files and did a basic validation similar to other ones for JSON etc. Let me know if this is fine.

File: pkg/validator/validator_test.go [hunk 1]
```diff
@@ -25,6 +25,34 @@ var testData = []struct {
 	{"validHcl", []byte(`key = "value"`), true, HclValidator{}},
 	{"invalidHcl", []byte(`"key" = "value"`), false, HclValidator{}},
 	{"multipleInvalidHcl", []byte(`"key1" = "value1"\n"key2"="value2"`), false, HclValidator{}},
+	{"validPlist", []byte(`<?xml version="1.0" encoding="UTF-8"?>
```
Comment (kehoecj) [inline pkg/validator/validator_test.go:28]: These strings are really long - recommend moving to a separate variable

**Review by kehoecj — APPROVED** [review]

**Comment by sandyydk** [comment]
Will do. I will update the PR. Thanks

File: pkg/validator/validator_test.go (hunk 1 above)
Comment (sandyydk) [reply pkg/validator/validator_test.go]: Moved them to an embedded file. Now it's more readable.

File: index.md [hunk 2]
```diff
@@ -50,7 +50,11 @@ How many deployments have you done that needed to be rolled back due to a missin
 * INI
 * Properties
 * HCL
+<<<<<<< HEAD
```
Comment (kehoecj) [inline index.md:53]: Looks like some merge conflicts got pushed - please take a look

File: pkg/validator/valid.plist
Comment (kehoecj) [inline pkg/validator/valid.plist:1]: I would prefer to keep test files out of the pkg directory. Would rather just use a few variables in the tests themselves than have a test file in the pkg dir.

**Comment by sandyydk** [comment]
Sure. Will update the PR for the same. I was thinking if we should go with
sub directories within validators to include these test files? If not will
go with constant itself.

**Comment by kehoecj** [comment]
> Sure. Will update the PR for the same. I was thinking if we should go with sub directories within validators to include these test files? If not will go with constant itself.

Would prefer constants for now

**Review by kehoecj — APPROVED** [review]

**Review by jd4235 — COMMENTED** [review]
Guess I failed to click the Submit review button.  I thought the comment would still show up.

File: pkg/validator/plist.go [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package validator
+
+import (
+	"bytes"
+
+	"howett.net/plist"
```
[the remark this answers is not in the document]
Comment (kehoecj) [reply pkg/validator/plist.go:6]: Added a task to #60 to specify that only Apple XML is supported. I'll take a look into the other plist types and see how widely used they are and if they are worth supporting.

**Pull request merged**